sys/linux/test/syz_kvm_setup_cpu_arm64

```diff
@@ -0,0 +1,24 @@
+#
+# requires: arch=arm64
+#
+r0 = openat$kvm(0, &AUTO='/dev/kvm\x00', 0x0, 0x0)
+r1 = ioctl$KVM_CREATE_VM(r0, 0xae01, 0x0)
+r2 = ioctl$KVM_CREATE_VCPU(r1, 0xae41, 0x0)
+r3 = ioctl$KVM_GET_VCPU_MMAP_SIZE(r0, 0xae04)
+r4 = mmap$KVM_VCPU(&(0x7f0000009000/0x1000)=nil, r3, 0x3, 0x1, r2, 0x0)
+# Perform two uexits. The first one is done via a code blob:
+#       d2802000        mov     x0, #0x100                      // #256
+#       f2bbbba0        movk    x0, #0xdddd, lsl #16
+#       f900001f        str     xzr, [x0]
+# , which assumes registers x24-28 are zeroes.
+# The second uexit is done via a syzos API command that sets uexit exit code to 0xaaaa.
+#
+syz_kvm_setup_cpu$arm64(r1, r2, &(0x7f0000e8a000/0x18000), &AUTO=[{0x0, &AUTO=[@code={0x1, 0x20, "002080d2a0bbbbf21f0000f9", 0xd65f03c0}, @uexit={0x0, 0x18, 0xaaaa}], 0x38}], 0x1, 0x0, 0x0, 0x0)
+ioctl$KVM_RUN(r2, 0xae80, 0x0)
+# Check that KVM_RUN was interrupted after the first uexit.
+#
+syz_kvm_run_expect_uexit$arm64(r4, 0x0)
+ioctl$KVM_RUN(r2, 0xae80, 0x0)
+# Check that KVM_RUN was interrupted after the second uexit.
+#
+syz_kvm_run_expect_uexit$arm64(r4, 0xaaaa)
```

Comment:
@dvyukov, for this pseudo-syscall the test runner complains there's no coverage (which is true, because it is purely userspace). Any ideas how to deal with that? Maybe we shouldn't warn about missing coverage in pseudo-syscalls?

Reply:
This pseudo-syscall is always preceded by `ioctl$KVM_RUN`. Perhaps it should contain `ioctl$KVM_RUN`? It would also solve the coverage problem.

Reply:
Yeah, we could fold it into KVM_RUN.

Reply:
I am thinking about adding `syz_kvm_run_loop$arm64(fd fd_kvmcpu, run kvm_run_ptr, allowed_exits int64, expect int64)` that would spin in a loop calling ioctl(fd_kvmcpu, KVM_RUN).

`allowed_exits` is a bitmask that lists the permitted types of VM exits. Depending on the exit type, we can check if the returned value is equal to `expect` and return -1 if they do not match:

```
// syz_kvm_run_loop$arm64(fd fd_kvmcpu, run kvm_run_ptr, allowed_exits int64, expect int64)
static long syz_kvm_run_loop(volatile long a0, volatile long a1, volatile long a2, volatile long a3) 
{
        const int cpufd = (int)a0;
        const struct kvm_run* run = (struct kvm_run*)a1;
        const uint64 allowed_exits = (uint64)a2;
        const uint64 expect = (uint64)a3;

        do {
                int ret = ioctl(cpufd, KVM_RUN, 0); 
                switch (run->exit_reason) {
                case KVM_EXIT_MMIO:
                        if (!(allowed_exits & (1 << KVM_EXIT_MMIO)))
                                continue;
                        if (run->mmio.phys_addr != ARM64_ADDR_UEXIT)
                                return -1; 
                        if (run->mmio.len != 8 || !run->mmio.is_write)
                                return -1; 
                        uint64 exit_code = *(uint64*)run->mmio.data;
                        return (exit_code == expect) ? 0 : -1; 
                }   
                return ret;
        } while (true);
}
```

Reply:
This will be test-only syscall, right?

Reply:
I don't see why it cannot be a prod syscall though.

Reply:
I think it will busy-loop in most cases (repeated wrong exit).

Reply:
Note that KVM_RUN busy-loops in most cases anyway. I think almost every auto-generated instruction sequence brings it to its knees.

But it looks like I am over-engineering this prematurely.
What if we just have `syz_kvm_run_till_uexit` now, which will execute KVM_RUN once and check the uexit code?
It will be usable in prod, but it won't be much different from ioctl$KVM_RUN.
And in the upcoming series that adds more functionality to syzos we can rework the pseudo-syscalls.

Reply:
I think we need to have 3-5 real diverse api calls before we can design testing. We are trying N-th iteration of testing, but I don't have full understanding of what exactly we are trying to test, which does not let me to understand if the approach is right and what are alternatives.

Reply:
The intent of this patch series was to get something roll on syzbot (e.g. try to reproduce bug 354724694) and iterate that. I have more API calls that I can add here, but I thought the review might get out of the hand if I bloat the series further.

Reply:
Resolving this per offline discussion.